src/element/resizeElements.ts

```diff
@@ -414,260 +386,193 @@ const resizeSingleTextElement = (
   }
 };
 
-const resizeSingleGenericElement = (
-  stateAtResizeStart: NonDeleted<ExcalidrawGenericElement>,
+const resizeSingleElement = (
+  stateAtResizeStart: NonDeletedExcalidrawElement,
   shouldKeepSidesRatio: boolean,
   element: NonDeletedExcalidrawElement,
   transformHandleDirection: TransformHandleDirection,
   isResizeFromCenter: boolean,
   pointerX: number,
   pointerY: number,
 ) => {
-  const [x1, y1, x2, y2] = getElementAbsoluteCoords(stateAtResizeStart);
+  // Gets bounds corners
+  const [x1, y1, x2, y2] = getResizedElementAbsoluteCoords(
+    stateAtResizeStart,
+    stateAtResizeStart.width,
+    stateAtResizeStart.height,
+  );
   const startTopLeft: Point = [x1, y1];
   const startBottomRight: Point = [x2, y2];
   const startCenter: Point = centerPoint(startTopLeft, startBottomRight);
 
   // Calculate new dimensions based on cursor position
-  let newWidth = stateAtResizeStart.width;
-  let newHeight = stateAtResizeStart.height;
   const rotatedPointer = rotatePoint(
     [pointerX, pointerY],
     startCenter,
     -stateAtResizeStart.angle,
   );
+
+  //Get bounds corners rendered on screen
+  const [esx1, esy1, esx2, esy2] = getResizedElementAbsoluteCoords(
```

Comment:
what does `es` prefix stand for?

Reply:
It's supposed to stand for "element on screen", but I see how that may be too cryptic :/ 

Any suggestions for improvement?

Reply:
Not sure what "element on screen" stands for even? :p — as opposed to what? Element off screen?

`boundsOnScreenWidth` — not clear either

Reply:
I clearly need to work on my naming skills xD

With "element on screen" I was trying to distinguish between the state of the element when the resizing started and the state of the element as shown to the user. 

So when I say something like "size on screen" I'm referring to the size of something that the user is currently seeing on the screen instead of when the resize started.

The other distinction I was trying to make was between the dimensions and coordinates of an element and the dimensions and coordinates of an element's bounding box. They coincide for generic elements, but they don't for non-generic ones.

So when I say something like the size of "bounds" I'm referring to the size of the bounding box and not the size of the element, which for non-generic elements differ.

Reply:
I'd just call it `boundsCurrentWidth` or `boundsCurrWidth`, as opposed to `initial*` or `original*` values.

Reply:
Sounds good. I'll make the changes.

Reply:
@dwelle, you meant `boundsCurrWidth` as opposed to `boundsOnScreen*` or as opposed to `initial*`?

It makes sense to me to replace `boundsOnScreen*` with `boundsCurr*` since it refers to the current element being displayed, but I can't quite make the connection if `boundsCurr*` is referring to the dimensions at the start of the resize (`initial*`)

Reply:
Yea I meant rename `boundsOnScreenWidth` to `boundsCurrWidth`. And `curr` would stand as opposite to `initial`.